OsmAnd/src/net/osmand/plus/mapcontextmenu/controllers/AmenityMenuController.java

```diff
@@ -101,11 +108,20 @@ public void buttonPressed() {
 
 	void openTrack(MapActivity mapActivity) {
 		TravelHelper travelHelper = mapActivity.getMyApplication().getTravelHelper();
-		String lang = amenity.getTagSuffix(Amenity.LANG_YES + ":");
-		String name = amenity.getTagContent(Amenity.ROUTE_NAME);
-		TravelArticle article = travelHelper.getArticleByTitle(name, lang, true, null);
-		if (article != null) {
-			travelHelper.openTrackMenu(article, mapActivity, name, amenity.getLocation());
+		if (ROUTE_ARTICLE_POINT.equals(amenity.getSubType())) {
+			String lang = amenity.getTagSuffix(Amenity.LANG_YES + ":");
+			String name = amenity.getTagContent(Amenity.ROUTE_NAME);
+			TravelArticle article = travelHelper.getArticleByTitle(name, lang, true, null);
+			if (article != null) {
+				travelHelper.openTrackMenu(article, mapActivity, name, amenity.getLocation(), false);
+			}
+		} else if (ROUTE_TRACK_POINT.equals(amenity.getSubType())) {
+			TravelGpx travelGpx = travelHelper.searchGpx(amenity.getLocation(), amenity.getRouteId(), amenity.getRef());
+			if (travelGpx != null) {
+				travelHelper.openTrackMenu(travelGpx, mapActivity, travelGpx.getTitle(), amenity.getLocation(), false);
+			} else {
+				LOG.error("openTrack() searchGpx() travelGpx is null");
```

Comment:
In these cases, it is better to add some info in logs about an object, for example + travelGpx.getTitle()

Reply:
a2cb935b28